boa/src/syntax/lexer/cursor.rs

```diff
@@ -70,31 +69,94 @@ where
+    where
+        F: Fn(char) -> bool,
+    {
+        Ok(match self.peek() {
+            None => false,
+            Some(Ok(peek)) => pred(*peek),
+            Some(Err(e)) => todo!(),
         })
     }
 
     /// Fills the buffer with all characters until the stop character is found.
     ///
     /// Note: It will not add the stop character to the buffer.
     pub(super) fn take_until(&mut self, stop: char, buf: &mut String) -> io::Result<()> {
-        unimplemented!()
+        loop {
+            if self.next_is(stop)? {
+                return Ok(());
+            } else {
+                match self.next() {
+                    None => {
+                        return Err(io::Error::new(
+                            ErrorKind::UnexpectedEof,
+                            format!("Unexpected end of file when looking for character {}", stop),
+                        ));
+                    }
+                    Some(Err(e)) => {
+                        return Err(e);
+                    }
+                    Some(Ok(ch)) => {
+                        buf.push(ch);
+                    }
+                }
+            }
+        }
+    }
+
+    /// Fills the buffer with characters until the first character (x) for which the predicate (pred) is false
+    /// (or the next character is none).
+    ///
+    /// Note that all characters up until x are added to the buffer including the character right before.
+    pub(super) fn take_until_pred<F>(&mut self, buf: &mut String, pred: &F) -> io::Result<()>
+    where
+        F: Fn(char) -> bool,
+    {
+        loop {
+            if !self.next_is_pred(pred)? {
+                return Ok(());
+            } else {
+                match self.next() {
+                    None => {
+                        unimplemented!();
+                        // unreachable!();
+                    }
+                    Some(Err(e)) => {
+                        return Err(e);
+                    }
+                    Some(Ok(ch)) => {
+                        buf.push(ch);
+                    }
+                }
+            }
+        }
     }
 
     /// Retrieves the given number of characters and adds them to the buffer.
-    pub(super) fn take(&mut self, count: usize, buf: &mut String) -> io::Result<()> {
+    pub(super) fn _take(&mut self, _count: usize, _buf: &mut String) -> io::Result<()> {
         unimplemented!()
     }
 
     /// It will fill the buffer with checked ASCII bytes.
-    pub(super) fn fill_bytes(&mut self, buf: &[u8]) -> io::Result<()> {
+    pub(super) fn fill_bytes(&mut self, _buf: &mut [u8]) -> io::Result<()> {
         unimplemented!()
     }
 
     /// Retrieves the next character as an ASCII character.
     ///
     /// It will make sure that the next character is an ASCII byte, or return an error otherwise.
-    pub(super) fn next_as_byte(&mut self) -> Option<io::Result<u8>> {
+    pub(super) fn _next_as_byte(&mut self) -> Option<io::Result<u8>> {
```

Comment:
We should not start function names with an underscore.

Reply:
Fixed